sdk/identity/azure-identity/samples/client_secret_credential.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <iostream>
+
+#include <azure/identity/client_secret_credential.hpp>
+
+#include <azure/service/client.hpp>
+
+// These functions should be getting the real Tenant ID, Client ID, and the Client Secret to
+// authenticate. It is recommended to NOT hardcode the secret in the code, but to get it from the
+// envoronment or read it from a secure location.
+std::string GetTenantId() { return std::string(); }
+std::string GetClientId() { return std::string(); }
+std::string GetClientSecret() { return std::string(); }
+
+int main()
+{
+  try
+  {
+    // Step 1: Initialize Cient Secret Credential.
+    auto clientSecretCredential = std::make_shared<Azure::Identity::ClientSecretCredential>(
+        GetTenantId(), GetClientId(), GetClientSecret());
+
+    // Step 2: Pass the credential to an Azure Service Client.
+    Azure::Service::Client azureServiceClient("some parameter", clientSecretCredential);
```

Comment:
oh I see, Azure::Service::Client is just a helper class to represent `any-azure-sdk-client`.

I think it would be  little more beneficial for customers to see a real-sdk-client on action here. Then, as a customer, I would be learning from this sample, things like:
- How to set up my CMake project to combine storage/keyvault client with Identity.
- How do I set up the client with the right credential.

Because, the first thing that came to my mind looking at these samples is:  "wait, do I need to create an SDK client first? "

Reply:
I think the ideal course of action is a separate repo with individual cmakefiles. Samples can be doing something useful from a real world, like multi-user messaging.
Here, this sample should work without taking a dependencty on any other service. When we write for `docs.microsoft.com`, we can take a real client such as Storage or KeyVault.
I'd leave it as is and left it to evolve.